govcd/catalog.go

```diff
@@ -351,52 +406,77 @@ func uploadOvfDescription(client *Client, ovfFile string, ovfUploadUrl *url.URL)
 	request := client.NewRequest(map[string]string{}, "PUT", *ovfUploadUrl, ovfReader)
 	request.Header.Add("Content-Type", "text/xml")
 
-	response, err := checkResp(client.Http.Do(request))
+	_, err = checkResp(client.Http.Do(request))
 	if err != nil {
 		return Envelope{}, err
 	}
 
 	var ovfFileDesc Envelope
-	ovfXml, err := ioutil.ReadAll(&buf)
-	if err != nil {
-		return Envelope{}, err
-	}
-
-	err = xml.Unmarshal(ovfXml, &ovfFileDesc)
+	err = parseOvfFileDesc(&buf, &ovfFileDesc)
 	if err != nil {
 		return Envelope{}, err
 	}
 
 	openedFile.Close()
 
-	body, err := ioutil.ReadAll(response.Body)
-	log.Printf("[TRACE] Response: %#v\n", response)
-	log.Printf("[TRACE] Response body: %s\n", string(body[:]))
-	log.Printf("[TRACE] Ovf file description file: %#v\n", ovfFileDesc)
+	return ovfFileDesc, nil
+}
 
-	response.Body.Close()
+func parseOvfFileDesc(buf *bytes.Buffer, ovfFileDesc *Envelope) error {
+	ovfXml, err := ioutil.ReadAll(buf)
+	if err != nil {
+		return err
+	}
 
-	return ovfFileDesc, nil
+	err = xml.Unmarshal(ovfXml, &ovfFileDesc)
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
 func findCatalogItemUploadLink(catalog *Catalog) (*url.URL, error) {
 	for _, item := range catalog.Catalog.Link {
 		if item.Type == "application/vnd.vmware.vcloud.uploadVAppTemplateParams+xml" && item.Rel == "add" {
-			log.Printf("[TRACE] Found Catalong link for uplaod: %s\n", item.HREF)
+			util.Logger.Printf("[TRACE] Found Catalong link for upload: %s\n", item.HREF)
 
 			uploadURL, err := url.ParseRequestURI(item.HREF)
 			if err != nil {
 				return nil, err
 			}
 
+			util.Logger.Printf("[TRACE] findCatalogItemUploadLink - upload url found: %s \n", uploadURL)
 			return uploadURL, nil
 		}
 	}
 	return nil, errors.New("catalog upload url isn't found")
```

Comment:
Message should be:  "Catalog upload URL not found"

Reply:
Fixed